lib/loader/index.js

```diff
@@ -71,12 +71,12 @@ function instantiate(module, imports) {
     const ptr0 = ptr >>> 2;
     const dataStart = (U32[ptr0] >>> 2) + 2; // 0 -> size; 1 -> static 0
     const len = U32[ptr0 + 1];
-    const recast = (x) => type.constructor.from(x);
-    if (type === U16 || type == U8) {
-      return recast(U32.subarray(dataStart, dataStart + len));
+
+    if (type === U8 || type === I8) {
+      return type.subarray(dataStart >>> 2, (dataStart >>> 2) + len)
```

Comment:
I8 and U8 don't need a shift, as each element is already 1 byte. Essentially, it's `>>> 1` for 16-bit, `>>> 2` for 32-bit and `>>> 3` for 64-bit values (`log2(byteSize)`).